kernel/src/upcall.rs

```diff
@@ -178,7 +183,8 @@ impl Upcall {
                 self.upcall_id.driver_num,
                 self.upcall_id.subscribe_num,
                 self.fn_ptr
-                    .map_or(core::ptr::null_mut::<()>(), |fp| fp.as_ptr()) as usize,
+                    .map_or(core::ptr::null_mut::<()>(), |fp| fp.as_ptr::<()>()
+                        as *mut ()) as usize,
```

Comment:
Style (nitpick): `.cast_mut()` here as well (if allowed by your Rust version).

Reply:
Just about exists, will change it!